lib/ansible/modules/cloud/digital_ocean/digital_ocean_droplet.py

```diff
@@ -0,0 +1,303 @@
+# Create a new Droplet
+# Will return the droplet details including the droplet id (used for idempotence)
+
+- digital_ocean_droplet:
+    state: present
+    name: mydroplet
+    oauth_token: XXX
+    size: 2gb
+    region: ams2
+    image: fedora-19-x64
+    wait_timeout: 500
+  register: my_droplet
+
+- debug:
+    msg: "ID is {{ my_droplet.droplet.id }}"
+
+- debug:
+    msg: "IP is {{ my_droplet.droplet.ip_address }}"
+
+# Ensure a droplet is present
+# If droplet id already exist, will return the droplet details and changed = False
+# If no droplet matches the id, a new droplet will be created and the droplet details (including the new id) are returned, changed = True.
+
+- digital_ocean_droplet:
+    state: present
+    id: 123
+    name: mydroplet
+    oauth_token: XXX
+    size: 2gb
+    region: ams2
+    image: fedora-19-x64
+    wait_timeout: 500
+'''
+
+
+RETURN = '''
+# Digital Ocean API info https://developers.digitalocean.com/documentation/v2/#droplets
+data:
+    description: a DigitalOcean Droplet
+    returned: changed
+    type: dict
+    sample: {
+        "droplet": {
+            "id": 3164494,
+            "name": "example.com",
+            "memory": 512,
+            "vcpus": 1,
+            "disk": 20,
+            "locked": true,
+            "status": "new",
+            "kernel": {
+                "id": 2233,
+                "name": "Ubuntu 14.04 x64 vmlinuz-3.13.0-37-generic",
+                "version": "3.13.0-37-generic"
+            },
+            "created_at": "2014-11-14T16:36:31Z",
+            "features": ["virtio"],
+            "backup_ids": [],
+            "snapshot_ids": [],
+            "image": {},
+            "volume_ids": [],
+            "size": {},
+            "size_slug": "512mb",
+            "networks": {},
+            "region": {},
+            "tags": ["web"]
+        }
+    }
+'''
+
+import time
+import traceback
+from ansible.module_utils._text import to_native
+from ansible.module_utils.basic import AnsibleModule, env_fallback
+from ansible.module_utils.digital_ocean import DigitalOceanHelper
+
+
+class DODroplet(object):
+    def __init__(self, module):
+        self.rest = DigitalOceanHelper(module)
+        self.module = module
+        self.wait = self.module.params.pop('wait', True)
+        self.wait_timeout = self.module.params.pop('wait_timeout', 300)
+        self.unique_name = self.module.params.pop('unique_name', False)
+        # pop the oauth token so we don't include it in the POST data
+        self.module.params.pop('oauth_token')
+
+    def get_by_id(self, droplet_id):
+        if not droplet_id:
+            return None
+        response = self.rest.get('droplets/{0}'.format(droplet_id))
+        json_data = response.json
+        if response.status_code == 200:
+            return json_data
+        return None
+
+    def get_by_name(self, droplet_name):
+        if not droplet_name:
+            return None
+        response = self.rest.get('droplets')
```

Comment:
You will need to implement links when performing this GET operation on droplets. Please take a look at the [links](https://developers.digitalocean.com/documentation/v2/#links) documentation.